gateway_conditions.md

```diff
@@ -0,0 +1,134 @@
+# GatewayClass & Gateway Status Conditions
+
+## GatewayClass conditions
+
+There is 1 condition type for GatewayClasses -  `Accepted`.
+
+### 'Accepted' GatewayClass condition type
+
+When a GatewayClass is created, it starts as below. This is the default defined in the CRD:
+
+```yaml
+    - lastTransitionTime: "1970-01-01T00:00:00Z"
+      status: Unknown
+      type: Accepted
+      reason: Waiting
+      message: Waiting for controller
+```
+
+When reconciled, the condition may change to 1 of 2 states.
+If the GatewayClass is valid *and* it is a supported class for which mctc will provision Gateways, the condition changes to:
+
+```yaml
+    - lastTransitionTime: "1970-01-01T00:00:00Z"
+      status: "True"
+      type: Accepted
+      reason: Accepted
+      message: Handled by kuadrant.io/mctc-gw-controller
+      observedGeneration: <corresponding metadata.generation>
+```
+
+If the Gateway is not valid or an unsupported class is specified, the condition changes to:
```

Comment:
I think we should be able to filter down to only gateways of a given class in the event handlers